src/lib/http.coffee

```diff
@@ -58,25 +51,32 @@ module.exports = (dnschain) ->
                 res.end()
                 cb()
 
-            resolver = switch req.headers.host
-                when 'icann.dns' then 'dns'
-                when 'namecoin.dns' then 'nmc'
-                when 'bitshares.dns' then 'bdns'
+            [...,resolverName] =
+                if S(req.headers.host).endsWith('.dns')
+                    S(req.headers.host).chompRight('.dns').s.split('.')
+                else if S(req.headers.blockchain).endsWith('.dns')
+                    S(req.headers.blockchain).chompRight('.dns').s.split('.')
                 else
-                    @log.warn gLineInfo "unknown host type: #{req.headers.host} -- defaulting to namecoin.dns!"
-                    'nmc'
+                    ['none']
+
+            if not (resolver = @dnschain.chains[resolverName])
+                @log.warn gLineInfo, 'unknown blockchain', {host: req.headers.host, blockchainHeader: req.headers.blockchain, remoteAddress: req.connection.remoteAddress}
```

Comment:
`gLineInfo` is a function that needs to be called (often with some argument, in this case `'unknown blockchain'` would be that argument). Double-check all your other uses of `gLineInfo` to make sure that you're doing it correctly.